bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/api/BondHttpApi.java

```diff
@@ -0,0 +1,295 @@
+        try {
+            return gson.fromJson(json, BondDeviceProperties.class);
+        } catch (JsonParseException e) {
+            logger.debug("Could not parse device {}'s property JSON '{}'", deviceId, json, e);
+            return null;
+        }
+    }
+
+    /**
+     * Executes a device action
+     *
+     * @param deviceId The ID of the device
+     * @param actionId The Bond action
+     * @param argument An additional argument for the actions (such as the fan speed)
+     */
+    public synchronized void executeDeviceAction(String deviceId, BondDeviceAction action, @Nullable Integer argument) {
+        String url = "http://" + bridgeHandler.getBridgeIpAddress() + "/v2/devices/" + deviceId + "/actions/"
+                + action.getActionId();
+        String payload = "{}";
+        if (argument != null) {
+            payload = "{\"argument\":" + argument + "}";
+        }
+        InputStream content = new ByteArrayInputStream(payload.getBytes());
+        try {
+            logger.debug("HTTP PUT to {} with content {}", url, payload);
+
+            final HttpClient httpClient = httpClientFactory.getCommonHttpClient();
+            final Request request = httpClient.newRequest(url).method(HttpMethod.PUT)
+                    .header("BOND-Token", bridgeHandler.getBridgeToken())
+                    .timeout(BOND_API_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+            try (final InputStreamContentProvider inputStreamContentProvider = new InputStreamContentProvider(
+                    content)) {
+                request.content(inputStreamContentProvider, "application/json");
+            }
+            ContentResponse response;
+            try {
+                response = request.send();
+            } catch (Exception e) {
+                throw new IOException(e);
+            }
+
+            logger.debug("HTTP response from {}: {}", deviceId, response.getStatus());
+        } catch (IOException ignored) {
+            logger.warn("Unable to execute device action!", ignored);
+        }
+    }
+
+    /**
+     * Submit GET request and return response, check for invalid responses
+     *
+     * @param uri: URI (e.g. "/settings")
+     */
+    private synchronized String request(String uri) throws IOException {
+        String httpResponse = "ERROR";
+        String url = "http://" + bridgeHandler.getBridgeIpAddress() + uri;
+        int numRetriesRemaining = 3;
+        do {
+            try {
+                logger.debug("HTTP GET to {}", url);
+
+                final HttpClient httpClient = httpClientFactory.getCommonHttpClient();
+                final Request request = httpClient.newRequest(url).method(HttpMethod.GET).header("BOND-Token",
+                        bridgeHandler.getBridgeToken());
+                ContentResponse response;
+                try {
+                    response = request.send();
+                } catch (Exception e) {
+                    throw new IOException(e);
```

Comment:
It seems strange to catch exception here, wrap it into `IOException` and then catch and unwrap it further down in same method. Could simply be caught directly instead of rethrown here.